galley/pkg/config/processor/transforms/ingress/gateway.go

```diff
@@ -60,7 +60,7 @@ func getGatewayXformProvider() transformer.Provider {
 
 func (g *gatewayXform) handle(e event.Event, h event.Handler) {
 
-	if g.options.MeshConfig.IngressControllerMode == meshconfig.MeshConfig_OFF {
+	if g.options.MeshConfig.IngressControllerMode == meshconfig.MeshConfig_OFF || e.Kind == event.Disabled {
```

Comment:
I think you need to handle this in the core logic of event.NewFnTransform, similar to how FullSync is handled. The transformers themselves (that use event.NewFnTransform) should not be depended to handle the Disabled event correctly.

Reply:
Got rid of this, obsolete.